### PR TITLE
Cleanup refactoring first commit

### PR DESCRIPTION
**Pull request by amen-souissi**
### Your checklist for this pull request

- [X] Make sure you are requesting to **pull a topic/feature/bugfix branch** (right side). Don't request your master!
- [X] Make sure you are making a pull request against the **master branch** (left side). Also you should start _your branch_ off _our master_.
- [X] Check the commit's or even all commits' message styles matches our requested structure.
- [X] Check your code additions will pass linting.

### Description
cleanup and refactoring, scroll management

### CODE REVIEW

File: src/components/scroller.tsx [hunk 1]
```diff
@@ -81,8 +81,22 @@ class Scroller extends React.Component<IScrollerProps> {
...
 
-  public componentDidUpdate() {
+  public componentDidUpdate(prevProps: IScrollerProps) {
     this.initializeMaxScrollValues();
+    // keep the relative scrollLeft value when the virtualWidth has been changed
+    if (this.scrollerContainer.current) {
+      const { scrollLeft } = this.scrollerContainer.current;
+      const { virtualWidth } = this.props;
+      if (prevProps.virtualWidth !== virtualWidth) {
+        const newLeft = scrollLeft - (prevProps.virtualWidth - virtualWidth) / 2;
```
Comment (hyacintheknobloch) [inline src/components/scroller.tsx:91]: Why do you divide it by two ?

File: src/components/scroller.tsx [hunk 2]
```diff
@@ -81,8 +81,22 @@ class Scroller extends React.Component<IScrollerProps> {
...
+        this.scrollToLeft(newLeft);
+      }
+      // @TODO keep the relative scrollTop value
```
Comment (hyacintheknobloch) [inline src/components/scroller.tsx:94]: You start by horizontal scroll and then you will do vertical scroll ?

File: src/components/utils/table.tsx [hunk 3]
```diff
@@ -109,29 +109,24 @@ export const computeRowStyle = (options: Nullable<IRowOptions>): React.CSSProper
...
-
-  for (let i = 0; i < fixedIndexes.length; i += 1) {
-    if (fixedIndexes[i + 1] - fixedIndexes[i] !== 1) {
-      return fixedIndexes[i] + 1;
+export const scrollIndexToGridIndex = (scrollIndex: number, hiddenIndexes: number[] = []) => {
+  let previousHiddenItems = 0;
+  let itemsToJump = 0;
+  const isHidden = hiddenIndexes.includes(scrollIndex);
+  for (let i = 0; i < hiddenIndexes.length; i += 1) {
+    const hiddenIndex = hiddenIndexes[i];
```
Comment (hyacintheknobloch) [inline src/components/utils/table.tsx:121]: I prefer to use forEach, it's easier to read, but I don't know if you can access to previousHiddenItems and ItemsToJump variables in the forEach callback scope.

File: src/components/virtualizer.tsx [hunk 4]
```diff
@@ -203,37 +202,12 @@ class Virtualizer extends React.Component<IVirtualizerProps, IState> {
...
-            setGridState();
-          }
-        } else {
-          setGridState();
-        }
-      } else {
-        setGridState();
-      }
+      const newRowsState = this.getVisibleRowsState(scrollTop);
+      // @ts-ignore
```
Comment (hyacintheknobloch) [inline src/components/virtualizer.tsx:209]: Why

File: test/components/table/utils.test.tsx [hunk 5]
```diff
@@ -42,25 +42,25 @@ describe("computeCellStyle", () => {
   });
 });
 
-describe("getFirstUnfixedIndex method", () => {
-  test("should return the first number (5) not present in the fixedIndexes list", () => {
-    const firstUnfixedIndex = Utils.getFirstUnfixedIndex([0, 1, 2, 3, 4]);
-    expect(firstUnfixedIndex).toEqual(5);
+describe.only("scrollIndexToGridIndex method", () => {
```
Comment (hyacintheknobloch) [inline test/components/table/utils.test.tsx:45]: only

File: src/components/scroller.tsx (hunk 2 above)
Comment (amen-souissi) [reply src/components/scroller.tsx:94]: yes

File: src/components/utils/table.tsx (hunk 3 above)
Comment (amen-souissi) [reply src/components/utils/table.tsx:121]: for loop it's faster than the forEach

File: src/components/scroller.tsx (hunk 1 above)
Comment (amen-souissi) [reply src/components/scroller.tsx:91]: the middle of the added item (only for 1 item, to improve for several items)

**Pull request merged**